quick_start/kusto_sample_app.py

```diff
@@ -20,394 +16,458 @@
+            print("Failed to alter the ingestion policy, which could be the result of insufficient permissions. The sample will still run, though ingestion will be delayed for up to 5 minutes.")
+
+    @classmethod
+    def create_ingestion_mappings(cls, use_existing_mapping:bool, kusto_client: KustoClient, database_name: str, table_name: str, mapping_name: str, mapping_value: str, data_format: DataFormat) -> bool:
+        if not use_existing_mapping:
+            if cls.data_format_to_is_mapping_required(data_format) and not mapping_value:
+                print(f"The data format '{data_format.name}' requires a mapping, but configuration indicates to not use an existing mapping and no mapping was provided. Skipping this ingestion.")
+                return False
+
+            if mapping_value:
+                ingestion_mapping_kind = cls.data_format_to_ingestion_mapping_type(data_format).value.lower()
+                cls.wait_for_user_to_proceed(f"Create a '{ingestion_mapping_kind}' mapping reference named '{mapping_name}'")
+
+                if not mapping_name:
+                    mapping_name = "DefaultQuickstartMapping" + UUID.randomUUID().toString().substring(0, 5)
+                mapping_command = f".create-or-alter table {table_name} ingestion {ingestion_mapping_kind} mapping '{mapping_name}' '{mapping_value}'"
+                if not cls.execute_control_command(kusto_client, database_name, mapping_command):
+                    print(f"Failed to create a '{ingestion_mapping_kind}' mapping reference named '{mapping_name}'. Skipping this ingestion.")
+                    return False
+        elif cls.data_format_to_is_mapping_required(data_format) and not mapping_name:
+            print(f"The data format '{data_format.name}' requires a mapping and the configuration indicates an existing mapping should be used, but none was provided. Skipping this ingestion.")
+            return False
 
+        return True
 
-def ingest_data_from_file(client: QueuedIngestClient, db: str, table: str, file_path: str, file_format: DataFormat, mapping_ref: str = None):
-    ingestion_props = IngestionProperties(
-        database=f"{db}",
-        table=f"{table}",
-        ingestion_mapping_reference=mapping_ref,
-        # Learn More: For additional information about supported data formats, see
-        #  https://docs.microsoft.com/azure/data-explorer/ingestion-supported-formats
-        data_format=file_format,
-        # Todo - Config: Setting the ingestion batching policy takes up to 5 minutes to have an effect.
-        #  For the sake of the sample we set Flush-Immediately, but in practice it should not be commonly used.
-        #  Comment the below line after running the sample for the first few times!
-        flush_immediately=True,
-    )
+    @classmethod
+    def ingest(cls, file: dict, data_format: DataFormat, ingest_client: QueuedIngestClient, database_name: str, table_name: str, mapping_name: str) -> None:

```

Comment:
Prefer a parent of `QueuedIngestClient`, which doesn't exist. Asaf is adding one, so we should change this once it merges.